src/nvim/channel.c

```diff
@@ -700,7 +700,7 @@ static void channel_process_exit_cb(Process *proc, int status, void *data)
   if (chan->term) {
     char msg[sizeof("\r\n[Process exited ]") + NUMBUFLEN];
     snprintf(msg, sizeof msg, "\r\n[Process exited %d]", proc->status);
```

Comment:
could move this `msg` construction into `terminal_close()` since we are passing the status now.